resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -827,13 +851,14 @@ private[spark] class MesosClusterScheduler(
       status: Int): Unit = {}
 
   private def removeFromQueuedDrivers(subId: String): Boolean = {
```

Comment:
Could you describe a little bit more about the reason why you change this function? The old logic looks identical to me.

Reply:
You are right: this is exactly the same logic! But the old implementation could not be kept that way due to the `queuedDrivers` collection whose type has changed from `ArrayBuffer` to `mutable.TreeSet`. In a `mutable.TreeSet`, you can remove an element by specifying the actual element, but you can't remove an element by its index.